src/ucp/core/ucp_am.c

```diff
@@ -214,6 +214,11 @@ static UCS_F_ALWAYS_INLINE int ucp_am_recv_check_id(ucp_worker_h worker,
     return 1;
 }
 
+static UCS_F_ALWAYS_INLINE size_t
+ucp_am_send_req_total_size(ucp_request_t *req) {
+    return req->send.length + req->send.msg_proto.am.header_length;
```

Comment:
maybe cache this value

Reply:
request is full in the corresponding enum